tests/formats/fork_choice/README.md

```diff
@@ -0,0 +1,101 @@
+# Fork choice tests
+
+The aim of the tests for the fork choice rules.
+
+## Test case format
+
+### `meta.yaml`
+
+```yaml
+description: string    -- Optional. Description of test case, purely for debugging purposes.
+bls_setting: int       -- see general test-format spec.
+```
+
+### `anchor_state.ssz_snappy`
+
+An SSZ-snappy encoded `BeaconState`, the state to initialize store with `get_forkchoice_store(anchor_state: BeaconState, anchor_block: BeaconBlock)` helper.
+
+### `anchor_block.ssz_snappy`
+
+An SSZ-snappy encoded `BeaconBlock`, the block to initialize store with `get_forkchoice_store(anchor_state: BeaconState, anchor_block: BeaconBlock)` helper.
+
+### `steps.yaml`
+
+The steps to execute in sequence. There may be multiple items of the following types:
+
+#### `on_tick` execution step
+
+The parameter that is required for executing `on_tick(store, time)`.
+
+```yaml
+{ tick: int }   --  to execute `on_tick(store, time)`
+```
+
+After this step, the `store` object may have been updated.
+
+#### `on_attestation` execution step
+
+The parameter that is required for executing `on_attestation(store, attestation)`.
+
+```yaml
+{ attestation: string }:  -- the name of the `attestation_<32-byte-root>.ssz_snappy` file. To execute `on_attestation(store, attestation)` with the given attestation.
+```
+The file is located in the same folder (see below).
+
+After this step, the `store` object may have been updated.
+
+#### `on_block` execution step
+
+The parameter that is required for executing `on_block(store, block)`.
+
+```yaml
+{ block: string }:        -- the name of the `block_<32-byte-root>.ssz_snappy` file. To execute `on_block(store, block)` with the given attestation.
+```
+The file is located in the same folder (see below).
+
+After this step, the `store` object may have been updated.
+
+#### Checks step
+
+The checks to verify the current status of `store` .
+
+```yaml
+checks: {<store_field_name>: value}   -- the assertions.
+```
+
+`<store_field_name>` is the field member of [`Store`](../../../specs/phase0/fork-choice.md#store) object that maintained by client implementation. Currently, the possible fields included:
+
+```yaml
+time: int                                   -- store.time
+genesis_time: int                           -- store.genesis_time
+justified_checkpoint_root: string           -- store.justified_checkpoint.root
+finalized_checkpoint_root: string           -- store.finalized_checkpoint_root.root
+best_justified_checkpoint_root: string      -- store.best_justified_checkpoint_root.root
```

Comment:
The example lists `head`, but it's a function instead of a field, and not listed here.